feat(datahub-frontend): add Horizontal Pod Autoscaler configuration

rafaelsmf opened the pull request:
### Description:

This PR implements Horizontal Pod Autoscaling (HPA) for the `datahub-frontend` component to address increasing resource consumption (CPU and memory) as the number of users on the platform grows. While increasing the number of replicas can help with resource usage, it doesn’t provide a long-term solution. By adding HPA, this implementation provides a more scalable and elegant approach to handle fluctuating loads, ensuring optimal performance and resource utilization.

How it works:

The HPA scales the `datahub-frontend` pods based on CPU and memory usage, automatically adjusting the number of pods according to demand.

It is configured via the `values.yaml` file, where enabling HPA for the frontend activates the autoscaling functionality.

### Testing:

The implementation was tested locally in a Minikube environment and in AWS EKS to verify proper functionality.

To test, enable HPA for the frontend in the `values.yaml` file and apply the changes to your cluster. The system will automatically scale based on resource usage.


## Checklist
- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [x] Docs related to the changes have been added/updated (if applicable)

pedro93 reviewed (commented):
Couple of notes.

I would recommend introducing some default stabilization logic in this HPA. Load on the frontend deployment (really any DataHub component) is quite spiky due to it's inherent nature. To prevent constant scale up/downs, add something like a stabilization window, see https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/#configurable-scaling-behavior 

Would also recommend adding some comments in the docs & values.yaml that this is a non-trivial behaviour change in the deploymenet if enabled that may case instability. I.e: The K8s node pool must be sized appropriately to ensure the HPA doesn't wait for node provisioning which depending on the backing system of K8s may take longer than the spike in usage lasts.

Please also bump the datahub-frontend subchart version as this PR introduces new functionality. This will also require updating the top-level datahub `chart.yaml` datahub-frontend dependency.

rafaelsmf commented:
Hey @pedro93, I added the documentation and updated the default behavior as well `datahub-frontend` version as you pointed out ([35bfe7d](https://github.com/acryldata/datahub-helm/pull/569/commits/35bfe7d01af6fa65702f54a6e8810dc933dd3f36) and [[1cf6b2a](https://github.com/acryldata/datahub-helm/pull/569/commits/1cf6b2a7b2c1f49aeb3efaede0945bf15c08c4dd)]).

pedro93 commented on charts/datahub/subcharts/datahub-frontend/values.yaml:
```diff
@@ -4,6 +4,26 @@
 
+hpa:
+  enabled: false # If enabled, Horizontal Pod Autoscaler (HPA) will automatically scale the number of replicas based on CPU and memory utilization thresholds.
+  minReplicas: 1
```
Usually recommend the default number of replicas in an HA (high availability) deployment to be 2.

This is important as it ensures the frontend deployment can always service requests even during rollouts such as programmed restarts, configuration changes & upgrades.

It's not the default value in `replicaCount` simply because the community usually use the defaults for testing, not for production ready set ups.  In this case it wouldn't make sense to have an HPA system scale the deployment to 1 if the replicaCount is set to 2 (recommended in production deployments).
